### DISCUSSION TITLE
Mermaid support - Github Enterprise Server

### DISCUSSION THREAD

**Discussion by mrkosima** [community/community#14551] [Enterprise] [answered]
Hello, GH recently shipped `mermaid` support for GH public and GH Cloud:
https://github.com/github/roadmap/issues/372

When do you plan to add the feature to GHES?

**Comment by davidstaheli** [comment, marked as answer]
Hi, @mrkosima! We plan to release Mermaid support in GHES in approximately November 2022. The release will appear on [GitHub's public roadmap](https://github.com/github/roadmap) the next time it is updated in early August. Thanks for your patience!

**Reply by ptasev** [reply]
What about math expression support for GHES?

https://github.com/orgs/community/discussions/17969

**Reply by davidstaheli** [reply]
Hi @ptasev. Math expression support should be included in that November 2022 GHES release too.

**Reply by mrmanc** [reply]
Excited for this to drop. It’s November, and I’m wondering if it’s just me or it hasn’t made it to the public roadmap yet? Do you have any update on when this is expected for GHES?

**Reply by mrkosima** [reply]
> Excited for this to drop. It’s November, and I’m wondering if it’s just me or it hasn’t made it to the public roadmap yet? Do you have any update on when this is expected for GHES?

It's rolled out a couple of weeks ago with **GHES 3.7.0**:
https://docs.github.com/en/enterprise-server@3.7/admin/release-notes

**Reply by mrmanc** [reply]
Amazing… need to have a conversation as we are soon to update to `3.6.3`, and this is a very compelling feature. Thanks for the info!

**Reply by divyaGrvgithub** [reply]
Hi, @mrkosima! We plan to release Mermaid support in GHES in approximately November 2022. The release will appear on [GitHub's public roadmap](https://github.com/github/roadmap) the next time it is updated in early August. Thanks for your patience!

**Reply by btoo** [reply]
Are there any plans to add mermaid support to the github ios app?

**Comment by ThomasJRyan** [comment]
Really looking forward to this update! It'll be nice to finally be able to graph our project layouts and get my team all on the same page

**Comment by matthewpersico** [comment]
+1

**Reply by dselman** [reply]
Great news.

**Comment by sarnobat** [comment]
Is there a discussion page anywhere that explains the criteria that Mermaid passed to become natively supported? I'd like to think 10-15 years down the line what kind of headaches I can expect if I put all my bets on Mermaid syntax as my primary diagramming engine.

**Reply by Kochise** [reply]
Count on Mermaid as de facto standard with little to no alternative.

[Kroki!](https://kroki.io/) would have been better though, because it supports way more "engines", including [Mermaid](https://mermaid.live/).

**Reply by sarnobat** [reply]
Thanks for the info.

I feel safer with plaintext output even though it's not as beautiful.
* https://github.com/weidagang/text-diagram/blob/master/README#L23
* https://github.com/ironcamel/Graph-Easy/wiki#hello-world

**Comment by Kochise** [comment]
Thank you all for your efforts, that's a great addition. Will `Graphviz` be supported as well someday ?

https://github.com/mdaines/viz.js

Online editors :

http://viz-js.com/

https://dreampuf.github.io/GraphvizOnline/

http://magjac.com/graphviz-visual-editor/

**Reply by abhishek171287** [reply]
Thank you everyone for the effort. Can anyone confirm GHES 3.8 supports mermaid?

**Reply by tdabasinskas** [reply]
It's already supported in GHES 3.7 - https://docs.github.com/en/enterprise-server@3.7/admin/release-notes:

> Users can use Mermaid syntax when writing Markdown, which displays a diagram when rendering the Markdown. For more information, see "[Creating diagrams](https://docs.github.com/en/enterprise-server@3.7/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams)."

**Reply by abhishek171287** [reply]
Yes. But we are planning to upgrade GHES to 3.8 ...does that support mermaid as well?

**Reply by Ryanflex2003** [reply]
Can anyone confirm it's working in GHES 3.7?  My current organization is on 3.7.4 and it does not render them, and I don't understand why since the docs clearly say it's supported as of 3.7.   I can take the same mermaid graph and copy it over to my personal Github and it renders just fine.  Thoughts?

**Comment by Kochise** [comment]
[Mermaid markdown rendering does not work in the Github Wiki](https://github.com/github/docs/issues/15727)

**Comment by nair-sumesh** [comment]
How to know mermaid version used by Github Enterprise Server?

**Reply by pchernikov-sxm** [reply]
We are trying to figure out the same

**Reply by ChristophHoyer** [reply]
good question, unfortunately no answer. :-/

**Reply by nair-sumesh** [reply]
Hi,
Anyone have any clue about this?
What could be done to upgrade mermaid to latest version in GHES?

**Reply by zimmerman-pliant** [reply]
To know the version, you can create a mermaid code block in an issue or document with just the word `info` in it:

````
```mermaid
  info
```
````

![image](https://github.com/user-attachments/assets/06f4bb99-34b9-413e-8aab-56851400723c)

Then in the preview it'll be replaced with the version number on your Github Enterprise server:

![image](https://github.com/user-attachments/assets/d9f4a5db-019e-492c-8d42-0ed81706ac46)

I'm not sure how the release cycles between GHES version and its Mermaid version work, or if you can get your site admin to upgrade Mermaid without upgrading GHES.

**Comment by divyaGrvgithub** [comment]
Hi, @mrkosima! We plan to release Mermaid support in GHES in approximately November 2022. The release will appear on [GitHub's public roadmap](https://github.com/github/roadmap) the next time it is updated in early August. Thanks for your patience!
